core/arch/arm/plat-sam/scmi_server.c

```diff
@@ -1116,6 +1117,46 @@ size_t plat_scmi_rd_count(unsigned int channel_id)
 	return resource->rd_count;
 }
 
+void sam_set_clock_range(unsigned int pmc_type, unsigned int pmc_id,
+			 const struct clk_range *output)
+{
+	struct sam_pmc_clk *p = pmc_clks;
+	unsigned int i = 0;
+
+	for (i = 0; i < ARRAY_SIZE(pmc_clks); i++, p++) {
+		if (pmc_type == p->pmc_type && pmc_id == p->pmc_id) {
+			p->output.min = output->min;
+			p->output.max = output->max;
+			return;
+		}
+	}
+}
+
+int32_t plat_scmi_clock_rates_by_step(unsigned int channel_id,
+				      unsigned int scmi_id,
+				      unsigned long *steps)
+{
+	const struct sam_pmc_clk *p = pmc_clks;
+	int32_t res = SCMI_NOT_SUPPORTED;
+	unsigned int i = 0;
+
+	if (channel_id == 0) {
+		for (i = 0; i < ARRAY_SIZE(pmc_clks); i++, p++) {
```

Comment:
nitpicking: `p` initialization should preferably be in the `for` instruction.
Maybe `for (p = pmc_clks; p < pmc_clks + ARRAY_SIZE(pmc_clks); p++) {` is enougth.
`